add logout feature (along with login)

wenjzhu opened the pull request:
## Description
logout with a clickable dropdown menu on nav bar. 

**Fixes #(add issue number here and remove parentheses)**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [ ] Within a week
- [x] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [x] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [ ] I have updated the [CHANGELOG](../CHANGELOG.md).
- [x] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [x] My changes generate no new warnings.
- [x] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

sandeep-ps commented on contributions/catalog/webapps/templates/contribute/contribute.html:
```diff
@@ -22,10 +20,10 @@
+        <a href="{{ url_for('contribute.create') }}"><i class="fa fa-plus-square fa-lg"></i> Contribute</a>
         <div style="float: right; margin-right: 25px;">
-            <a >Hello, {{ user }}!</a>
+            <a ><i class="fa fa-fw fa-user fa-lg"></i> Hello, {{ user }}!</a>
```
Please use the pointer cursor when hovering over the name/icon to make it clear that it'sclickable. Thanks.

wenjzhu replied:
updated.

sandeep-ps commented:
@wenjzhu, If I'm on the Create Contibution page, the logout option is not appearing.  Also, please add both issues to the CHANGELOG and resolve the conflict. Thanks

wenjzhu commented:
> @wenjzhu, If I'm on the Create Contibution page, the logout option is not appearing. Also, please add both issues to the CHANGELOG and resolve the conflict. Thanks

I fixed the logout function for Create Contribution page. Also, I added the feature to only allow logged-in user to see the create contribution page.

sandeep-ps commented on contributions/catalog/controllers/contribute.py:
```diff
@@ -4,25 +4,47 @@
+
+@bp.route('/login', methods=['GET', 'POST'])
+def login():
+    print("login.")
```
Please remove the print statements and use logger debug if needed. Thanks.

sandeep-ps commented:
Hi @wenjzhu, Basic functionality-wise, this looks good. There are some print statements in the PR that could be removed. Please use the logger debug option instead if needed. Thanks.

wenjzhu commented:
> Hi @wenjzhu, Basic functionality-wise, this looks good. There are some print statements in the PR that could be removed. Please use the logger debug option instead if needed. Thanks.

just updated.
